### PR TITLE
Add Object Priority Mode Register

### PR DESCRIPTION
**Pull request by rbong**
See: https://gbdev.io/pandocs/CGB_Registers.html?highlight=ff6c#ff6c--opri-cgb-mode-only-object-priority-mode

### CODE REVIEW

**Comment by CasualPokePlayer** [comment]
This register effect on priority is only effective when the bootrom is unmapped [(or if one of the PGB modes are active, according to SameBoy's source)](https://github.com/LIJI32/SameBoy/blob/237e127b36eaee574e2a1d7900da21bda0775405/Core/memory.c#L1397). Once those conditions are no longer met then any more writes will not effect priority, and the register effectively just becomes a free bit of RAM.

I would assume this would be useful for homebrew bootrom authors, but otherwise perhaps the docs on this should note it is (mostly?) bootrom only?

**Comment by rbong** [comment]
> This register effect on priority is only effective when the bootrom is unmapped [(or if one of the PGB modes are active, according to SameBoy's source)](https://github.com/LIJI32/SameBoy/blob/237e127b36eaee574e2a1d7900da21bda0775405/Core/memory.c#L1397). Once those conditions are no longer met then any more writes will not effect priority, and the register effectively just becomes a free bit of RAM.
> 
> I would assume this would be useful for homebrew bootrom authors, but otherwise perhaps the docs on this should note it is (mostly?) bootrom only?

Thanks, I added a short comment about that. I missed that from Pandocs.

**Review by avivace — APPROVED** [review]

File: hardware.inc [hunk 1]
```diff
@@ -795,6 +796,19 @@ DEF rOCPD EQU $FF6B
...
+; -- OPRI ($FF6C)
+; -- Object Priority Mode (R/W)
+; -- CGB Mode Only
+; --
+; -- Priority can be changed only from the boot ROM
+; --
+DEF rOPRI EQU $FF6C
+
+DEF OPRI_OAM   EQU 0 ; Prioritize objects by location in OAM (CGB default)
+DEF OPRI_COORD EQU 1 ; Prioritize objects by x-coordinate (DMG default)
```
Comment (ISSOtm) [inline hardware.inc:809]:
```suggestion
DEF OPRI_OAM   EQU 0 ; Prioritize objects by location in OAM (CGB Mode default)
DEF OPRI_COORD EQU 1 ; Prioritize objects by x-coordinate (Non-CGB Mode default)

```

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Thanks!

**Pull request merged**